encode.go

```diff
@@ -75,11 +76,28 @@ func emitCopy(dst []byte, offset, length int) int {
 	return i
 }
 
+var encPool = sync.Pool{New: func() interface{} { return new(encoder) }}
+
 // Encode returns the encoded form of src. The returned slice may be a sub-
 // slice of dst if dst was large enough to hold the entire encoded block.
 // Otherwise, a newly allocated slice will be returned.
 // It is valid to pass a nil dst.
 func Encode(dst, src []byte) []byte {
+	var e = encPool.Get().(*encoder)
+	dst = e.Encode(dst, src)
```

Comment:
Defer is slow, and there is no need for it. If there were complex return paths, it would be something else.